src/pixelator/plot/__init__.py

```diff
@@ -791,6 +791,129 @@ def plot_colocalization_diff_heatmap(
     return plt.gcf(), plt.gca()
 
 
+def _add_top_marker_labels(
+    differential_colocalization, n_top_pairs: int = 5, min_log_p: float = 5.0
+):
+    differential_colocalization = differential_colocalization.sort_values(
+        "median_difference"
+    )
+    differential_colocalization = differential_colocalization.loc[
+        -np.log10(differential_colocalization["p_adj"]) > min_log_p, :
+    ]
+
+    ## Labels for marker pair withs highest negative differential colocalization scores
+    for _, row in differential_colocalization.head(n_top_pairs).iterrows():
+        x, y = row[["median_difference", "p_adj"]]
+        y = -np.log10(y)
+        if x > 0:
+            continue
+        label = plt.text(
```

Comment:
Would it be possible to pass a specific plot object to these and then add the text to that? I feel a bit uncomfortable with this depending on global state from `plt` here.

Reply:
Good point! now fixed.